src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5693,7 +5693,8 @@ bool MergeTreeData::supportsLightweightDelete() const
     auto lock = lockParts();
     for (const auto & part : data_parts_by_info)
     {
-        if (!part->supportLightweightDeleteMutate())
+        if (part->getState() == MergeTreeDataPartState::Active
```

Comment:
I think we should also return false for these states (not 100% sure though)
```
    Temporary,       /// the part is generating now, it is not in data_parts list
    PreActive,       /// the part is in data_parts, but not used for SELECTs
```

Maybe it's better to inverse the check and explicitly skip parts in Deleting and Outdated states like this:

```
     for (const auto & part : data_parts_by_info)
     {
        if (part->getState() == DataPartState::Outdated ||
            part->getState() == DataPartState::Deleting)
            continue;

         if (!part->supportLightweightDeleteMutate())
             return false;
     }
```

Reply:
I think, you are right.